api2/pkg/webapp/templates/individual.gohtml

```diff
@@ -240,18 +260,127 @@
         }
 
+        // Returns the secondPartyType ID corresponding to the selection of the relationship type
+        // dropdown
+        const getSecondPartyType = () => {
+            const relationshipTypeSelector = document.getElementById("relationshipTypeSelector")
+            let selectedRelationshipType = relationshipTypes.find(
+                e => e.id === relationshipTypeSelector.value
+                    && e.firstPartyRole === relationshipTypeSelector.options[relationshipTypeSelector.selectedIndex].text
+            )
+            return selectedRelationshipType.secondPartyType
+        }
+
+        // Returns the value of the partySelector search box
+        const getSearchParam = () => {
+            const partySelector = document.getElementById("partySelector")
+            return partySelector.textContent
+        }
+
+        // Party List Item click handler - sets the text box value used for searching to the selected item
+        // from the dropdown
+        const onPartyListItemClick = (event) => {
+            document.getElementById("partySelector").value = event.target.textContent
+            document.getElementById("partySelector").setAttribute("data-party-id", event.target.id)
+        }
+
+        // Get First Name from party attribute list, or return blank
+        const firstNameFromAttrs = (attrs) => {
+            let firstName = ""
+            Object.keys(attrs).forEach(
+                k => {
+                    if (k == "{{.FirstNameAttribute.ID}}") {
+                        firstName = attrs[k]
+                    }
+                }
+            )
+            return firstName
+        }
+
+        // Get Last Name from party attribute list, or return blank
+        const lastNameFromAttrs = (attrs) => {
+            let lastName = ""
+            Object.keys(attrs).forEach(
+                k => {
+                    if (k == "{{.LastNameAttribute.ID}}") {
+                        lastName = attrs[k]
+                    }
+                }
+            )
+            return lastName
+        }
+
         // On document ready
         document.addEventListener('DOMContentLoaded', function () {
 
             // Bind add relationship button
             document.getElementById('add-relationship-button').onclick = () => {
                 let typeSelector = document.getElementById('relationshipTypeSelector');
-                let partySelector = document.getElementById('relationshipSecondPartySelector');
+                let partySelector = document.getElementById('partySelector');
                 let relationshipType = typeSelector.value;
-                let party = partySelector.value;
+                let party = partySelector.getAttribute("data-party-id");
                 addRelationship(relationshipType, party);
             };
 
+            // Add listener for party selector item click
+            Array.from(document.getElementsByClassName('partySelectorItem')).forEach(e => {
```

Comment:
Do we need Array.from?

Reply:
Yes, because getElementsByClassName returns an HTMLCollection which cannot directly be iterated over with Array methods. To be able to use .forEach on it, I needed to change it to an Array first.

See here: [JS: iterating over result of getElementsByClassName using Array.forEach](https://stackoverflow.com/questions/3871547/js-iterating-over-result-of-getelementsbyclassname-using-array-foreach)